src/ncp/ncp_base_radio.cpp

```diff
@@ -434,28 +445,38 @@ otError NcpBase::HandlePropertySet_SPINEL_PROP_STREAM_RAW(uint8_t aHeader)
     otRadioFrame *frame;
 
     VerifyOrExit(otLinkRawIsEnabled(mInstance), error = OT_ERROR_INVALID_STATE);
+    if (otLinkRawIsTransmittingOrScanning(mInstance))
+    {
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+        ExitNow(error = EnqueuePendingCommand(kPendingCommandTypeTransmit, aHeader, 0));
+#else
+        ExitNow(error = OT_ERROR_INVALID_STATE);
+#endif
+    }
+    else
+    {
+        frame = otLinkRawGetTransmitBuffer(mInstance);
+        VerifyOrExit(frame != nullptr, error = OT_ERROR_NO_BUFS);
 
-    frame = otLinkRawGetTransmitBuffer(mInstance);
-    VerifyOrExit(frame != nullptr, error = OT_ERROR_NO_BUFS);
-
-    SuccessOrExit(error = DecodeStreamRawTxRequest(*frame));
+        SuccessOrExit(error = DecodeStreamRawTxRequest(*frame));
 
-    // Cache the transaction ID for async response
-    mCurTransmitTID = SPINEL_HEADER_GET_TID(aHeader);
+        // Pass frame to the radio layer. Note, this fails if we
+        // haven't enabled raw stream or are already transmitting.
+        SuccessOrExit(error = otLinkRawTransmit(mInstance, &NcpBase::LinkRawTransmitDone));
 
-    // Pass frame to the radio layer. Note, this fails if we
-    // haven't enabled raw stream or are already transmitting.
-    error = otLinkRawTransmit(mInstance, &NcpBase::LinkRawTransmitDone);
+        // Cache the interface and transaction ID for async response
+        mCurTransmitIID = SPINEL_HEADER_GET_IID(aHeader);
+        mCurTransmitTID = SPINEL_HEADER_GET_TID(aHeader);
+    }
 
 exit:
-
-    if (error == OT_ERROR_NONE)
+    if (error != OT_ERROR_NONE)
     {
-        // Don't do anything here yet. We will complete the transaction when we get a transmit done callback
-    }
-    else
-    {
-        error = WriteLastStatusFrame(aHeader, ThreadErrorToSpinelStatus(error));
+        // If we fail to report the error now, it will be reported later in HandleReceive() of ncp_base.cpp.
+        if (WriteLastStatusFrame(aHeader, ThreadErrorToSpinelStatus(error)) == OT_ERROR_NONE)
+        {
+            return OT_ERROR_NONE;
```

Comment:
This help ensure we still have single `return` in each function/method (which I think is required by our style guide).
```suggestion
            error = OT_ERROR_NONE;
```